NewsApp/NewsApp/Controller/SearchController.swift

```diff
@@ -9,40 +9,182 @@
+        }
     }
 
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+
+        guard let parametersRequest = sender as? ParametersRequest else {
+            return
+        }
+
         if segue.identifier == R.segue.searchController.segueToNews.identifier {
             guard let newsController = segue.destination as? NewsController else {
                 return
             }
-            guard let parametersRequest = sender as? ParametersRequest else {
-                return
-            }
             newsController.viewTitle = parametersRequest.query
             newsController.newsService = NewsServiceImpl(parametersRequest: parametersRequest)
-            searchBar.resignFirstResponder()
+        } else if segue.identifier == R.segue.searchController.segueToSearchDateFrom.identifier {
+            guard let searchDateFromController = segue.destination as? SearchDateFromController else {
+                return
+            }
+            searchDateFromController.delegate = self
+        } else if segue.identifier == R.segue.searchController.segueToSearchDateTo.identifier {
+            guard let searchDateToController = segue.destination as? SearchDateToController else {
+                return
+            }
+            searchDateToController.delegate = self
+        } else if segue.identifier == R.segue.searchController.segueToSearchLanguage.identifier {
+            guard let searchLanguageController = segue.destination as? SearchLanguageController else {
+                return
+            }
+            searchLanguageController.delegate = self
         }
+        searchBar.resignFirstResponder()
     }
 }
-//TODO: Un seul mot doit être tapé
+
+///Initialisation du controleur
+extension SearchController {
+    func initSearchController() {
+        self.navigationItem.title = R.string.localizable.search()
+        self.searchBar.placeholder = R.string.localizable.searchForAnArticle()
+        self.sortByLabel.text = R.string.localizable.sortBy()
+        self.languageLabelWithTitle.text = R.string.localizable.language()
+        self.dateFromLabelWithTitle.text = R.string.localizable.startDate()
+        self.dateToLabelWithTitle.text = R.string.localizable.endDate()
+        self.searchButton.setTitle(R.string.localizable.searchVerb(), for: .normal)
+        self.moreFiltersButton.setTitle(R.string.localizable.moreFilters(), for: .normal)
+        initSegmentedControl()
+        initFilters()
+    }
+
+    func initSegmentedControl() {
+        for sortByIndice in 0..<self.sortBy.count {
+            self.sortBySegmentedControl.setTitle(self.sortBy[sortByIndice].title, forSegmentAt: sortByIndice)
+        }
+    }
+
+    func initFilters() {
+        if let dateFrom = self.parametersRequest.from {
+            self.dateFromLabelWithDate.text = dateFrom.formattedStringforDisplay
+        } else {
+            self.dateFromLabelWithDate.text = R.string.localizable.selectADate()
```

Comment:
Amélioration d'écriture possible avec ??

Reply:
fix in "fix PR filtres recherche"